pandas/core/indexes/multi.py

```diff
@@ -324,20 +324,26 @@ def from_arrays(cls, arrays, sortorder=None, names=None):
                    codes=[[0, 0, 1, 1], [1, 0, 1, 0]],
                    names=['number', 'color'])
         """
+        error_msg = "Input must be a list / sequence of array-likes."
         if not is_list_like(arrays):
-            raise TypeError("Input must be a list / sequence of array-likes.")
+            raise TypeError(error_msg)
         elif is_iterator(arrays):
             arrays = list(arrays)
 
         # Check if lengths of all arrays are equal or not,
         # raise ValueError, if not
         for i in range(1, len(arrays)):
+            if not is_list_like(arrays[i]):
+                raise TypeError(error_msg)
```

Comment:
Hmm what was this doing before if not raising this message already?

Reply:
it raises `TypeError: object of type 'int' has no len()` if it's an `int`